### ISSUE TITLE
esbuild cannot resolve npm prefixed packages

### ISSUE THREAD

**Issue by ryanleecode** [paritytech/capi#838]
```
running examples/sign/ed25519.eg.ts
✘ [ERROR] Do not know how to load path: npm:@noble/ed25519

    examples/sign/ed25519.eg.ts:11:20:
      11 │ import * as ed from "npm:@noble/ed25519"
         ╵                     ~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] Do not know how to load path: npm:@noble/hashes/sha512

    examples/sign/ed25519.eg.ts:12:23:
      12 │ import { sha512 } from "npm:@noble/hashes/sha512"
         ╵                        ~~~~~~~~~~~~~~~~~~~~~~~~~~

running examples/paginate.eg.ts
error: Uncaught Error: Build failed with 2 errors:
examples/sign/ed25519.eg.ts:11:20: ERROR: Do not know how to load path: npm:@noble/ed25519
examples/sign/ed25519.eg.ts:12:23: ERROR: Do not know how to load path: npm:@noble/hashes/sha512
  let error = new Error(`${text}${summary}`);
              ^
    at failureErrorWithLog (https://deno.land/x/esbuild@v0.17.12/mod.js:1612:15)
    at https://deno.land/x/esbuild@v0.17.12/mod.js:1024:25
    at runOnEndCallbacks (https://deno.land/x/esbuild@v0.17.12/mod.js:1447:45)
    at buildResponseToResult (https://deno.land/x/esbuild@v0.17.12/mod.js:1022:7)
    at https://deno.land/x/esbuild@v0.17.12/mod.js:1051:16
    at responseCallbacks.<computed> (https://deno.land/x/esbuild@v0.17.12/mod.js:673:9)
    at handleIncomingPacket (https://deno.land/x/esbuild@v0.17.12/mod.js:728:9)
    at readFromStdout (https://deno.land/x/esbuild@v0.17.12/mod.js:649:7)
    at https://deno.land/x/esbuild@v0.17.12/mod.js:1898:11
```

**Comment by tjjfvi** [comment]
This example should probably be reverted to use the esm.sh version

**Comment by harrysolovay** [comment]
I'll take this on

**Comment by ryanleecode** [comment]
Closed by #846
